cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java

```diff
@@ -51,49 +49,55 @@ public ExecutionTaskPlanner() {
     _executionId = new AtomicLong(0L);
     _partMoveProposalByBrokerId = new HashMap<>();
     _remainingReplicaMovements = new TreeSet<>();
-    _dataToMove = 0L;
+    _remainingDataToMove = 0L;
     _leaderMovements = new HashMap<>();
   }
 
-  public void addBalancingProposals(Collection<BalancingAction> proposals) {
-    for (BalancingAction proposal : proposals) {
-      addBalancingProposal(proposal);
+  public void addExecutionProposals(Collection<ExecutionProposal> proposals) {
+    for (ExecutionProposal proposal : proposals) {
+      addExecutionProposal(proposal);
     }
   }
 
   /**
    * Add a new proposal that needs to be executed.
    *
+   * A proposal will have at least one of the two following actions:
+   * 1. Replica action (i.e. movement, addition, deletion or order change).
+   * 2. Leader action (i.e. leader movement)
+   *
    * @param proposal the proposal to execute.
    */
-  public void addBalancingProposal(BalancingAction proposal) {
+  public void addExecutionProposal(ExecutionProposal proposal) {
+    // Ideally we should avoid adjust replica order if not needed, but due to a bug in open source Kafka
```

Comment:
Is there an open ticket that we can point to?

Reply:
It is fixed in latest Kafka.